### PR TITLE
feat: cookie consent popup

### PR DESCRIPTION
**Pull request by uniqueg**

### CODE REVIEW

**Comment by samyak-max** [comment]
Hey, @git-anurag-hub just letting you know that the old feature PR for cookie consent have been closed; I have left some comments [there](https://github.com/elixir-cloud-aai/landing-page/pull/73); you can check them out and let me know for the required changes 🫡.

File: components/CookieConsent.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
+import pluginConfig from "./CookieConsentConfig";
+import "vanilla-cookieconsent";
+
+const CookieConsent = () => {
+  useEffect(() => {
+    /**
```
Comment (git-anurag-hub) [inline components/CookieConsent.js:7]: Remove Comments!

File: components/CookieConsentConfig.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          {
+            title: "Performance and Analytics cookies",
+            description:
+              "These cookies allow the website to remember the choices you have made in the past",
+            toggle: {
+              value: "analytics", // your cookie category
+              enabled: false,
+              readonly: false,
+            },
+            cookie_table: [
```
Comment (git-anurag-hub) [inline components/CookieConsentConfig.js:95]: I don't think right now we use any such cookie!

File: components/CookieConsentConfig.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  // cookie_same_site: 'Lax',                // default: 'Lax'
+  // use_rfc_cookie: false,                  // default: false
+  // revision: 0,                            // default: 0
+
+  onFirstAction: function (user_preferences, cookie) {
+    // callback triggered only once
+  },
+
+  onAccept: function (cookie) {
+    // ...
```
Comment (git-anurag-hub) [inline components/CookieConsentConfig.js:30]: Any documentation on what this function do! As right now I am not sure it is handling any action(s).

File: components/CookieConsentConfig.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  },
+
+  onChange: function (cookie, changed_preferences) {
+    // ...
```
Comment (git-anurag-hub) [inline components/CookieConsentConfig.js:34]: Any documentation on what this function do! As right now I am not sure it is handling any action(s).

**Review by git-anurag-hub — CHANGES_REQUESTED** [review]
Hey @samyak-max, I am not sure if you saw but you won't be able to modify the css settings of this library you are using easily/efficiently. I will prefer using a headless library so that we can modify the UI/UX of the cookie-consent popup according to our requirements or the best/preferred way will be to create your own cookie consent popup & use a library to like [cookies-next ](https://www.npmjs.com/package/cookies-next) for managing the cookies server-side.

**Comment by git-anurag-hub** [comment]
Also @samyak-max, please see if you can also resolve the conflicts. In case on any help regarding anything feel free to DM me. :)

File: components/CookieConsentConfig.js (hunk 3 above)
Comment (samyak-max) [reply components/CookieConsentConfig.js:30]: image

This function isn't handling any action as of now, this function is for loading extra scripts.

File: components/CookieConsentConfig.js (hunk 4 above)
Comment (samyak-max) [reply components/CookieConsentConfig.js:34]: image

This function is also not handling any actions.

**Comment by samyak-max** [comment]
> Hey @samyak-max, I am not sure if you saw but you won't be able to modify the css settings of this library you are using easily/efficiently. I will prefer using a headless library so that we can modify the UI/UX of the cookie-consent popup according to our requirements or the best/preferred way will be to create your own cookie consent popup & use a library to like [cookies-next ](https://www.npmjs.com/package/cookies-next) for managing the cookies server-side.

Actually, this library was linked in the [issue](https://github.com/elixir-cloud-aai/landing-page/issues/51), so I preassumed that this is only to be used, sorry for that 😓. I'll create a new cookie consent popup using cookie-next 👍.

**Comment by uniqueg** [comment]
> > Hey @samyak-max, I am not sure if you saw but you won't be able to modify the css settings of this library you are using easily/efficiently. I will prefer using a headless library so that we can modify the UI/UX of the cookie-consent popup according to our requirements or the best/preferred way will be to create your own cookie consent popup & use a library to like [cookies-next ](https://www.npmjs.com/package/cookies-next) for managing the cookies server-side.
> 
> Actually, this library was linked in the [issue](https://github.com/elixir-cloud-aai/landing-page/issues/51), so I preassumed that this is only to be used, sorry for that 😓. I'll create a new cookie consent popup using cookie-next 👍.

I'm sorry about that. I just linked the first library I found. I personally don't care too much about the consent popup having w different site than the rest of the page. This GDPR stuff is a cancer anyway 😅

I think if that's the only remaining issue we can merge. Then you could open another issue to try with a headless library or the React server side library.

**Review by git-anurag-hub — APPROVED** [review]

**Comment by git-anurag-hub** [comment]
LGTM! @samyak-max @uniqueg

**Comment by samyak-max** [comment]
Hey @git-anurag-hub, I removed the onChange, onAccept, and onFirstAction functions as they didn't do anything for now. The cookies were saved by clicking the accept button itself. I also removed the "analytics" and "targeting and advertisement" cookies. The only cookie stored now is the 'necessary' cookie, I also moved the pluginConfig to the component itself.

image

**Comment by samyak-max** [comment]
> Good stuff, @samyak-max. Before I squash merge and delete your branch, can you maybe briefly document how to re-add the different sections and event handlers when we will have additional cookies in the future? It will be relevant for @tanya17-05 soon, as she's started implementing Google Analytics. If this is well described in the library docs, just point to the relevant sections. Otherwise just add some bullet points that may save @tanya17-05 some time, since you already know.
> 
> Thanks 🙏

Sure, I will do that 👍

**Comment by uniqueg** [comment]
Thanks a lot. I think it's best if you write it either here or in the corresponding issue. Please tell me when you are done so I can merge this one :)

**Comment by samyak-max** [comment]
This cookie consent uses the [vanilla-cookieconsent](https://www.npmjs.com/package/vanilla-cookieconsent) npm library. The cookies can be configured in the component itself by editing the pluginConfig object in the component itself. For now, there is only one "[necessary cookie](https://github.com/elixir-cloud-aai/landing-page/pull/79#issuecomment-1385750958)" stored, more can be created certainly, also cookie table can be created by adding cookie_table obj, and its headers modified by cookie_table_headers in settings_modal obj. The cookies are stored after clicking accept/reject, no need for any function call.

Apart from that many callbacks are available, 
- onChange function is valid if we allow the user to change their preference.  
- onAccept function runs when cookies are accepted and after reloads.
- onFirstAction function runs only on the initial choice made.
- page_scripts is for managing existing scripts (already set true).

Other callbacks are available on the npm page.

There are some API methods also available, they can be accessed under CookieConsentApi obj, list of available methods is also present on the npm page. Some data-cc actions are also present. 

PS there is a section for configuration with gtag (under Full example configurations), that might be useful.

**Pull request merged**